provisioning/e2e/_provisioning_e2e.js

```diff
@@ -43,25 +48,82 @@ var createX509Certificate = function (certOptions, callback) {
   });
 };
 
+var createIntermediateCaCert = function(authorityName, parentCert, callback) {
+  var certOptions = {
+    commonName: authorityName,
+    serviceKey: parentCert.key,
+    serviceCertificate: parentCert.cert,
+    serviceKeyPassword: '123',
```

Comment:
>serviceKeyPassword: '123', [](start = 4, length = 26)

We really need these?

Reply:
if we do, let's random-generate the password.

---
In reply to: [161123387](https://github.com/Azure/azure-iot-sdk-node/pull/198#discussion_r161123387) [](ancestors = 161123387)

Reply:
We don't actually need it.  This was copy/paste from other code that used a password-protected key.

---
In reply to: [161131200](https://github.com/Azure/azure-iot-sdk-node/pull/198#discussion_r161131200) [](ancestors = 161131200,161123387)

Reply:
Sorry, wrong file.  We do need this.  The cert that Jenkins has in it's environment variable has a password on its key.  

---
In reply to: [161300406](https://github.com/Azure/azure-iot-sdk-node/pull/198#discussion_r161300406) [](ancestors = 161300406,161131200,161123387)